qiime2/sdk/result.py

```diff
@@ -143,9 +143,41 @@ def export_data(self, output_dir):
     def _destructor(self):
         return self._archiver._destructor
 
-    def save(self, filepath):
-        if not filepath.endswith(self.extension):
-            filepath += self.extension
+    def save(self, filepath, ext=None):
+        """Save an artifact (.qza) or visualization (.qzv) file.
```

Comment:
Let's make this more generic, I think it might be a little confusing to read about Visualizations when saving an Artifact and vice versa. Maybe something like "Save to a file." is enough?